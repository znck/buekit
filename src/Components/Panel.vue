<script>
import { style, styleResolver } from '../utils'
import { createTag } from '../mixins'

export default {
  name: 'Panel',
  functional: true,
  mixins: [createTag('nav')],
  props: {
    title: { type: String }
  },
  render (h, ctx) {
    const s = styleResolver(ctx.$style)
    const { tag, title } = ctx.props
    const { title: slot, default: content = [] } = ctx.slots()

    return style(
      h(tag, ctx.data, [
        h('p', { class: s('panel-heading') }, slot || title),
        ...content
      ]),
      s('panel')
    )
  }
}
</script>

<style lang="scss" module>
@import '~bulma/sass/utilities/_all.sass';
@import '../_base.scss';
@import '~bulma/sass/components/panel.sass';
</style>
