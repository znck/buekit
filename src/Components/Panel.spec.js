import { shallow } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'

import Panel from './Panel.vue'

describe('Panel.vue', () => {

  it ('matches snapshot', () => {
    const wrapper = shallow(Panel, {})
    const renderer = createRenderer()

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
