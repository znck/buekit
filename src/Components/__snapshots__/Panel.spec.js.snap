// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel.vue matches snapshot 1`] = `
<nav class="panel">
    <p class="panel-heading"></p>
</nav>
`;
